### PR TITLE
[Constraint.Lagrangian] Remove merge option from BilateralInteractionConstraint

### PR DESCRIPTION
**Pull request by epernod**
based on PR #3327 

- Remove option `merge` which was added as a test in 2011 and never finished. According to a quick test on example. The behavior is less stable when activating the option with the same speed.

**Result of BilateralConstraint between 2 deformable models.** 
Current behavior: single constraint working fine
RemovingBilateralInteractionConstraint_00000001

Result when activating merge option: Some vibration and constraint is duplicated
RemovingBilateralInteractionConstraint_00000002

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.inl [hunk 1]
```diff
@@ -46,16 +46,9 @@ BilateralInteractionConstraint<DataTypes>::BilateralInteractionConstraint(Mechan
...
     , d_numericalTolerance(initData(&d_numericalTolerance, 0.0001, "numericalTolerance",
                                     "a real value specifying the tolerance during the constraint solving. (optional, default=0.0001)") )
-
-    //TODO(dmarchal): Such kind of behavior shouldn't be implemented in the component but externalized in a second component or in a python script controlling the scene.
-    , activateAtIteration( initData(&activateAtIteration, 0, "activateAtIteration", "activate constraint at specified interation (0 = always enabled, -1=disabled)"))
-
-    //TODO(dmarchal): what do TEST means in the following ? should it be renamed (EXPERIMENTAL FEATURE) and when those Experimental feature will become official feature ?
-    , merge(initData(&merge,false, "merge", "TEST: merge the bilateral constraints in a unique constraint"))
-    , derivative(initData(&derivative,false, "derivative", "TEST: derivative"))
+    , d_activate( initData(&d_activate, true, "activate", "controle constraint activation (true by default)"))
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.inl]:
```suggestion
    , d_activate( initData(&d_activate, true, "activate", "control constraint activation (true by default)"))
```

**Review by alxbilger — CHANGES_REQUESTED** [review]
I recommend to deprecate Data fields properly. See https://www.sofa-framework.org/community/doc/contributing-to-sofa/lifecycle-mechanisms/#removing-a-datafield

**Comment by epernod** [comment]
> I recommend to deprecate Data fields properly. See https://www.sofa-framework.org/community/doc/contributing-to-sofa/lifecycle-mechanisms/#removing-a-datafield

I did something to correctly depreciate the Data. The instruction are not 100% clear to me (regarding dates inside the macro and parse method). Let me know if it seems correct.

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.h [hunk 2]
```diff
@@ -124,7 +121,24 @@ class BilateralInteractionConstraint : public PairInteractionConstraint<DataType
...
+    /// Temporary function to warn the user when old attribute names are used
+    void parse(sofa::core::objectmodel::BaseObjectDescription* arg) override
+    {
+        Inherit::parse(arg);
+
+        if (arg->getAttribute("activateAtIteration"))
+        {
+            msg_warning() << "input Data 'activateAtIteration' has been deprecated, please use 'activate' instead and an engine or a script to change the behavior at the right step (see PR #3327).";
+        }
+        else if (arg->getAttribute("merge"))
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.h]:
```suggestion
        if (arg->getAttribute("merge"))
```

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.h [hunk 3]
```diff
@@ -124,7 +121,24 @@ class BilateralInteractionConstraint : public PairInteractionConstraint<DataType
...
+        {
+            msg_warning() << "input Data 'merge' has been deprecated. Its behavior was unused, undocumented, untested, and unclear (see PR #3328).";
+        }
+        else if (arg->getAttribute("derivative"))
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/BilateralInteractionConstraint.h]:
```suggestion
        if (arg->getAttribute("derivative"))
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Pull request merged**